sparkle/solver/verifiers.py

```diff
@@ -66,5 +70,56 @@ def sat_judge_correctness_raw_result(instance: Path,
         return SATVerifier.sat_get_verify_string(sat_verify.stdout.decode())
 
 
+class SolutionFileVerifier(SolutionVerifier):
+    """Class to handle the file verifier."""
+
+    def __init__(self: SolutionFileVerifier, csv_file: Path) -> None:
+        """Initialize the verifier by building dictionary from csv.
+
+        Args:
+            csv_file: path to the csv file. Requires lines to be of the form:
+                instance,objective,solution
+        """
+        self.csv_file = csv_file
+        lines = [line.split(",", maxsplit=2)
+                 for line in self.csv_file.read_text().splitlines()]
+        self.solutions = {instance: (objective, solution)
+                          for instance, objective, solution in lines}
+
+    def __str__(self: SATVerifier) -> str:
+        """Return the name of the SAT verifier."""
+        return SolutionFileVerifier.__name__
+
+    def verify(self: SolutionFileVerifier,
+               instance: Path,
+               solver_output: dict[str, object],
+               solver_call: list[str]) -> SolverStatus:
+        """Verify the solution.
+
+        Args:
+            instance: instance to verify, solution found by instance name as key
+            solver_output: outcome of the solver to verify
+
+        Returns:
+            The status of the solver on the instance
+        """
+        instance = instance.name
+        if instance not in self.solutions:
+            return SolverStatus.UNKNOWN
+
+        objective, solution = self.solutions[instance]
+        if objective not in solver_output:
+            return SolverStatus.UNKNOWN
+
+        outcome = solver_output[objective]
+        if solution != outcome:
```

Comment:
The solution in the file might be "UNKOWN", in such case any solution the solver returns is acceptable (this can happen if we configure the solver and it is now able to solve an instance that was unsolvable before)

Reply:
Hmmm this seems like an interesting edge case. So you're saying the solution file maybe does not hold all the solutions?

Reply:
Handled in latest push, but the edge case where the instance is not present in the solutions will still end up in UNKNOWN, would 'original' answer of solver be preferable?